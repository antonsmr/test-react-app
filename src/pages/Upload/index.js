import React, { useEffect, useState } from 'react';
import { connect } from 'react-redux';
import { Upload, Button, notification } from 'antd';
import { isEmpty, uniqBy } from 'lodash';
import { InboxOutlined, SmileOutlined } from '@ant-design/icons';
import { shape, func } from 'prop-types';

import Layout from '../../components/Layout';
import { toBase64 } from '../../utils/base64Parser';

import {
  uploadVideo,
} from '../../actions/upload';

const { Dragger } = Upload;

const UploadPage = (props) => {
  const [fileList, setFileList] = useState([]);

  const { upload } = props;

  const openNotification = () => {
    notification.open({
      message: 'File uploader',
      description:
        'Video files were successfully uploaded.',
      icon: <SmileOutlined style={{ color: '#108ee9' }} />,
    });
  };

  useEffect(() => {
    if (props.upload.uploaded) {
      openNotification();
      setFileList([]);
    }
  }, [props.upload.uploaded]);

  const handleUpload = async () => {
    const promises = await fileList.map(async (file) => {
      const base64String = await toBase64(file);
      return {
        name: file.name,
        base64String,
      };
    });
    const mappedList = await Promise.all(promises);

    props.uploadVideo(mappedList);
  };

  const beforeUpload = (file, files) => {
    setFileList(uniqBy([...fileList, ...files], 'uid'));
    return false;
  };

  const onRemove = (file) => {
    const index = fileList.indexOf(file);
    const newFileList = [...fileList].slice();
    newFileList.splice(index, 1);
    setFileList(newFileList);
  };

  return (
    <Layout>
      <h4>Upload a video</h4>
      <div className="content-upload">
        <Dragger fileList={fileList} beforeUpload={beforeUpload} onRemove={onRemove} multiple>
          <p className="ant-upload-drag-icon">
            <InboxOutlined />
          </p>
          <p className="ant-upload-text">Click or drag file to this area to upload</p>
          <p className="ant-upload-hint">
            Support for a single or bulk upload. Strictly prohibit from uploading company data or
            other band files
          </p>
        </Dragger>
        <Button
          className="btn-upload"
          loading={upload.uploading}
          disabled={isEmpty(fileList)}
          onClick={handleUpload}
        >
          Upload
        </Button>
      </div>
    </Layout>
  );
};

UploadPage.propTypes = {
  upload: shape().isRequired,
  uploadVideo: func.isRequired,
};
UploadPage.defaultProps = {};

const mapStateToProps = (state) => ({
  upload: state.upload,
});

const actionCreators = {
  uploadVideo,
};

export default connect(mapStateToProps, actionCreators)(UploadPage);
