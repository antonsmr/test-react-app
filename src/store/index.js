import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import { createLogger } from 'redux-logger';
import thunk from 'redux-thunk';
import combinedReducers from '../reducers';
import config from '../config';

export const configureStore = () => {
  const logger = createLogger({
    collapsed: (getState, action, logEntry) => !logEntry.error,
  });

  const create = () => {
    if (config.env === 'development') {
      return createStore(combinedReducers, composeWithDevTools(
        applyMiddleware(
          thunk,
          logger,
        ),
      ));
    }

    return createStore(combinedReducers, applyMiddleware(thunk));
  };

  const store = create();

  if (module.hot) {
    module.hot.accept('../reducers', () => {
      store.replaceReducer(combinedReducers);
    });
  }

  return store;
};

export default configureStore;
