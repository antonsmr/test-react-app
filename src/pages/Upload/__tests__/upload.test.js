import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Upload from '../index';
import { configureStore } from '../../../store';

import '@testing-library/jest-dom';

configure({ adapter: new Adapter() });

const mockProps = {
  upload: {
    uploading: false,
    uploaded: false,
  },
  uploadVideo: jest.fn(),
};

const store = configureStore();

describe('Upload page', () => {
  it('should be defined', () => {
    expect(Upload).toBeDefined();
  });

  test('should run example code', () => {
    expect(mockProps.upload).toBeDefined;
    expect(mockProps.uploadVideo).toBeDefined;
    expect(mockProps.uploadVideo).toMatchSnapshot();
  });

  it('shallow render test', () => {
    shallow(
      <Router>
        <Provider store={store}>
          <Upload {...mockProps} />
        </Provider>
      </Router>,
    );
  });
});
