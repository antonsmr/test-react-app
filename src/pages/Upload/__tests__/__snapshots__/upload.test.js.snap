// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Upload page should run example code 1`] = `[MockFunction]`;
